### ISSUE TITLE
3.19.2: Trace events from Microsoft.IdentityModel.Clients.ActiveDirectory.AdalOption.AdalEventSource seem to be coming as Errors when they should be Information or Verbose

### ISSUE THREAD

**Issue by fpintos** [AzureAD/azure-activedirectory-library-for-dotnet#1039]
**Issue**
Events from AdalOption.AdalEventSource are reaching the event listener with name & level of "Error", when they should in fact be informational events.

This is likely due to [DefaultLog](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/130219b656485b4f536be62af1a18c4e123fe09d/adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Features/WinCommon/AdalLogger.cs) not taking the level into account and just calling the event source Error().

**Expected**
Events to come with correct level and be filtered as expected by the event listener.

**Details/Repro**
I have an event listener hooked up to AdalOption.AdalEventSource that looks like this:
```csharp
    public class TracingEventListener : EventListener
    {
        protected override void OnEventWritten(EventWrittenEventArgs eventData)
        {
            var message = GetMessage(eventData);
            Trace.WriteLine($"level={eventData.Level},{message}");
        }

        protected static string GetMessage(EventWrittenEventArgs eventData)
        {
            if (eventData.Message != null)
            {
                return string.Format(eventData.Message, eventData.Payload.ToArray());
            }

            var sb = new StringBuilder($"eventId={eventData.EventId},eventName={eventData.EventName},");
            sb.Append('(');
            for (var i = 0; i < eventData.PayloadNames.Count; i++)
            {
                sb.Append(eventData.PayloadNames[i]);
                sb.Append('=');
                sb.Append((eventData.Payload[i] ?? "<null>"));
                if (i != eventData.PayloadNames.Count - 1)
                {
                    sb.Append(',');
                }
            }

            sb.Append(')');
            return sb.ToString();
        }
    }
```

And it is initialized with:
```csharp
var eventListener = new TracingEventListener();
eventListener.EnableEvents(AdalOption.AdalEventSource, EventLevel.Error);
```

When I run my application I get lots of events like this:
```
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:02.8477150Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: ADAL PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:02.8497206Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: Client
        , Authority Host: login.windows.net)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0532637Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Loading from cache.)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0582786Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0612850Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Looking up cache for a token...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:03.0622878Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: No matching token was found in the cache)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:04.6690196Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: Storing token in the cache...)
 level=Error,eventId=4,eventName=Error,(message=2018-04-17T19:27:04.6700221Z: c3d7238e-eb72-4a4b-9b7e-c533d3808ffb - LoggerBase.cs: An item was stored in the cache)
```

In the event listener EventWrittenEventArgs shows the events with Error level as well.

If it helps, here is a list of other packages in the app (it is a test project):
```xml
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="FluentAssertions" version="5.2.0" targetFramework="net46" />
  <package id="Microsoft.Azure.KeyVault" version="2.3.2" targetFramework="net46" />
  <package id="Microsoft.Azure.KeyVault.WebKey" version="2.0.7" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="3.19.2" targetFramework="net46" />
  <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net46" developmentDependency="true" />
  <package id="Microsoft.OData.Client" version="7.4.2" targetFramework="net46" />
  <package id="Microsoft.OData.Core" version="7.4.2" targetFramework="net46" />
  <package id="Microsoft.OData.Edm" version="7.4.2" targetFramework="net46" />
  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net46" />
  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.10" targetFramework="net46" />
  <package id="Microsoft.Spatial" version="7.4.2" targetFramework="net46" />
  <package id="MSTest.TestAdapter" version="1.2.0" targetFramework="net46" />
  <package id="MSTest.TestFramework" version="1.2.0" targetFramework="net46" />
  <package id="Newtonsoft.Json" version="11.0.1" targetFramework="net46" />
  <package id="System.ValueTuple" version="4.4.0" targetFramework="net46" />
</packages>
```

A sample callstack where the event listener is invoked goes like this:
```
 	Microsoft.Bookings.OData.Service.FunctionalTests.TracingEventListener.OnEventWritten(System.Diagnostics.Tracing.EventWrittenEventArgs = {System.Diagnostics.Tracing.EventWrittenEventArgs}) Line 19	C#
 	System.Diagnostics.Tracing.EventSource.DispatchToAllListeners(int = 4, System.Guid* = 0x00000000, System.Diagnostics.Tracing.EventWrittenEventArgs = {System.Diagnostics.Tracing.EventWrittenEventArgs}) Line 2192	C#
 	System.Diagnostics.Tracing.EventSource.WriteToAllListeners(int, System.Guid*, object[]) Line 2166	C#
 	System.Diagnostics.Tracing.EventSource.WriteToAllListeners(int, System.Guid*, int, System.Diagnostics.Tracing.EventSource.EventData*) Line 2151	C#
 	System.Diagnostics.Tracing.EventSource.WriteEventWithRelatedActivityIdCore(int = 4, System.Guid*, int, System.Diagnostics.Tracing.EventSource.EventData*) Line 1307	C#
 	System.Diagnostics.Tracing.EventSource.WriteEvent(int, string) Line 877	C#
>	Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform.Logger.DefaultLog(Microsoft.IdentityModel.Clients.ActiveDirectory.LogLevel, string)	Unknown
 	Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform.LoggerBase.Log(Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.CallState, Microsoft.IdentityModel.Clients.ActiveDirectory.LogLevel = Information, string, bool, string)	Unknown
 	Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform.LoggerBase.Information(Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.CallState, string)	Unknown
 	Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.AcquireTokenHandlerBase(Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.RequestData = {Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.RequestData})	Unknown
```

**Comment by jennyf19** [comment]
Thanks for the detailed information @fpintos 
Are you using 3.19.2?

**Comment by jennyf19** [comment]
With this fix, @fpintos verified that the events come with the correct level and can be filtered as expected by the event listener.

**Comment by jennyf19** [comment]
Addressed with [3.19.4](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/releases/tag/v3.19.4) release

@fpintos
